File: src/Wcs.Plc.Entities/Plc.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Wcs.Plc.Entities
{
  [Table("plcs")]
  public class Plc
  {
    [Key]
    public virtual int id { get; set; } = 0;

    [Required]
    public virtual string model { get; set; } = "test";

    [Required]
    public virtual string name { get; set; }

    public virtual string host { get; set; } = "";

    public virtual int port { get; set; } = 0;

    public virtual bool is_running { get; set; } = false;

    [Required]
    public virtual DateTime created_at { get; set; } = DateTime.Now;

    [ForeignKey("plc_id")]
    public List<PlcState> states { get; set; }

  }
}

File: src/Tiantong.Iot/Tiantong.Iot.Client/src/entities/Plc.ts
import DateTime from '@/share/utils/DateTime'

export class Plc {
  id = 0
  name = ''
  number = ''
  model = PlcModel.mc3eBinary
  host = '192.168.1.1'
  port = 8000
  comment = ''
  created_at = DateTime.now
}

export enum PlcModel {
  mc3eBinary = 'mc3e-binary',
  s7200Smart = 's7200smart',
  test = 'test',
}
